c2corg_ui/static/js/feed.js

```diff
@@ -0,0 +1,153 @@
+goog.provide('app.FeedController');
+goog.provide('app.feedDirective');
+
+goog.require('app');
+goog.require('app.Api');
+goog.require('app.Authentication');
+goog.require('app.utils');
+
+
+/**
+ * @return {angular.Directive} The directive specs.
+ */
+app.feedDirective = function() {
+  return {
+    restrict: 'A',
+    controller: 'appFeedController as feedCtrl'
+  };
+};
+app.module.directive('appFeed', app.feedDirective);
+
+
+/**
+ * @param {angular.Scope} $scope Scope.
+ * @param {angular.Attributes} $attrs Attributes.
+ * @param {app.Authentication} appAuthentication
+ * @param {app.Api} appApi Api service.
+ * @param {angular.$sce} $sce Angular Strict Contextual Escaping
+ * @constructor
+ * @ngInject
+ * @export
+ */
+app.FeedController = function($scope, $attrs, appAuthentication, appApi, $sce) {
+
+  /**
+   * @type {angular.Scope}
+   * @private
+   */
+  this.scope_ = $scope;
+
+  /**
+   * @type {app.Api}
+   * @private
+   */
+  this.api_ = appApi;
+
+  /**
+   * @type {app.Authentication}
+   * @private
+   */
+  this.auth_ = appAuthentication;
+
+  /**
+   * @type {angular.$sce} $sce Angular Strict Contextual Escaping
+   * @private
+   */
+  this.sce_ = $sce;
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.location_ = $attrs['location'];
+
+  /**
+   * @type {Array<Object>}
+   * @export
+   */
+  this.documents = [];
+
+  /**
+   * @type {string | undefined}
+   * @private
+   */
+  this.nextToken_;
+
+  /**
+   * @type {boolean}
+   * @export
+   */
+  this.busy = true;
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.feedType_;
+
+  switch (this.location_) {
```

Comment:
I wonder if ``this.location_`` and ``this.feedType_`` are not a bit duplicates.

What about dropping ``location`` and use only ``app-feed-type`` in the HTML.
In the homepage, this attribute would be omitted and
* if is auth => personal
* if not => standard
In a user profile we would add ``app-feed-type="profile"`` to the directive.